testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/micrometer/MicrometerSetupTask.java

```diff
@@ -6,66 +6,59 @@
 
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.STATISTICS_ENABLED;
 
+import org.jboss.arquillian.testcontainers.api.DockerRequired;
+import org.jboss.arquillian.testcontainers.api.Testcontainer;
 import org.jboss.as.arquillian.container.ManagementClient;
 import org.jboss.as.controller.client.helpers.Operations;
 import org.jboss.as.test.shared.ServerReload;
 import org.jboss.as.test.shared.util.AssumeTestGroupUtil;
 import org.jboss.dmr.ModelNode;
+import org.junit.AssumptionViolatedException;
 import org.wildfly.test.integration.observability.container.OpenTelemetryCollectorContainer;
 import org.wildfly.test.integration.observability.setuptask.AbstractSetupTask;
 
 /**
  * Sets up a functioning Micrometer subsystem configuration. Requires functioning Docker environment! Tests using this
  * are expected to call AssumeTestGroupUtil.assumeDockerAvailable(); in a @BeforeClass.
  */
+@DockerRequired(AssumptionViolatedException.class)
 public class MicrometerSetupTask extends AbstractSetupTask {
-    protected boolean dockerAvailable = AssumeTestGroupUtil.isDockerAvailable();
+    private static final ModelNode micrometerExtension = Operations.createAddress("extension", "org.wildfly.extension.micrometer");
+    private static final ModelNode micrometerSubsystem = Operations.createAddress("subsystem", "micrometer");
 
-    private final ModelNode micrometerExtension = Operations.createAddress("extension", "org.wildfly.extension.micrometer");
-    private final ModelNode micrometerSubsystem = Operations.createAddress("subsystem", "micrometer");
-    private boolean extensionAdded = false;
-    private boolean subsystemAdded = false;
+    @Testcontainer
     private OpenTelemetryCollectorContainer otelCollector;
 
     @Override
     public void setup(final ManagementClient managementClient, String containerId) throws Exception {
+        AssumeTestGroupUtil.assumeDockerAvailable();
+
```

Comment:
This should not be required since `@DockerRequired(AssumptionViolatedException.class)` is on the task.